### ISSUE TITLE
Proxmox: upload ISO via API [Feature request]

### ISSUE THREAD

**Issue by lukasmrtvy** [hashicorp/packer#7855]
Would be super useful if Packer /as it can serve kickstart file/ also can upload iso file.

Possible with curl:
```
 curl  -X POST "$URL/api2/json/nodes/$NODE/storage/$STORAGE/upload" \
                -b "PVEAuthCookie=$AUTH" \
                -H "CSRFPreventionToken: $TOKEN" \
                -H "Content-Type: multipart/form-data" \
                --form "content=iso" \
                --form "filename=@$ISO_LOCATION" \
```
And possible delete after provisioning:
```
HTTP: | DELETE /api2/json/nodes/{node}/storage/{storage}/content/{volume}
```

**Comment by lukasmrtvy** [comment]
ping @carlpett

**Comment by carlpett** [comment]
Yep, sounds useful. None of the other builders I've used does it though...
@SwampDragons Does this sound like a feature which you want in Packer, or is it out of scope? Are there any other builders that do it?

**Comment by SwampDragons** [comment]
Just to clarify the request here -- would it basically be providing an ISO url, which Packer obtains and then uploads to the Proxmox host and then uses that iso to launch the VM?  

We do perform something similar for the VMWare-esxi builder, so it wouldn't be without precedent.  You can take a look at the implementation in https://github.com/hashicorp/packer/blob/master/builder/vmware/iso/step_remote_upload.go

**Comment by carlpett** [comment]
This needs to be implemented in the proxmox-api-go package first. I've started doing so, once that work is done it can be added to packer.

**Comment by carlpett** [comment]
Sorry for lack of updates here. I got bogged down quite a lot at work this autumn and didn't have time to finish this work. I resumed it tonight though, and now have a branch where it works. However, I found more work needed in the api package... Submitted a PR for that [here](https://github.com/Telmate/proxmox-api-go/pull/73), and once that is merged I can make a PR on packer.
